src/main/java/com/suse/saltstack/netapi/config/ClientConfig.java

```diff
@@ -12,6 +12,28 @@
     public static final Key<URI> URL = new Key<>(URI.create("http://localhost:8000"));
     public static final Key<String> TOKEN = new Key<>();
 
+    /**
+     * Timeout in milliseconds until a connection is established.
+     * A timeout of zero is interpreted as an infinite timeout.
+     * A negative value is interpreted as undefined (system default).
+     * Default value is 10000ms (10s)
+     * 
+     * @see HttpClientConnection.request(String)
+     * @see JDKConnection.request(String, String)
+     */
+    public static final Key<Integer> CONNECT_TIMEOUT = new Key<>(10000);
+
+    /**
+     * Timeout in milliseconds for waiting for data.
+     * A timeout of zero is interpreted as an infinite timeout.
+     * A negative value is interpreted as undefined (system default).
+     * Default value is 10000ms (10s)
+     * 
+     * @see HttpClientConnection.request(String)
+     * @see JDKConnection.request(String, String)
+     */
+    public static final Key<Integer> SOCKET_TIMEOUT = new Key<>(10000);
```

Comment:
I don't think we need extra constants for the default values of keys since they are already pretty self explanatory constants accessible through `SOCKET_TIMEOUT.defaultValue`.
Besides that they may have the same default value now but that could change.